appservice/src/intTest/groovy/nu/yona/server/DeleteDirectMessageTest.groovy

```diff
@@ -0,0 +1,365 @@
+				bobDunnBuddyMessageAcceptURL = response.responseData._embedded.buddyConnectRequestMessages[0]._links.accept.href
+				bobDunnBuddyConnectRequestDeleteURL = response.responseData._embedded.buddyConnectRequestMessages[0]._links.self.href
+			}
+
+		then:
+			response.status == 200
+			response.responseData._links.self.href == bobDunnURL + appService.DIRECT_MESSAGE_PATH_FRAGMENT
+			response.responseData._embedded.buddyConnectRequestMessages[0].user.firstName == "Richard ${timestamp}"
+			response.responseData._embedded.buddyConnectRequestMessages[0]._links.self.href.startsWith(response.responseData._links.self.href)
+			bobDunnBuddyMessageAcceptURL.startsWith(response.responseData._embedded.buddyConnectRequestMessages[0]._links.self.href)
+			bobDunnBuddyConnectRequestDeleteURL != null
+	}
+
+	def 'Bob tries to delete Richard\'s buddy request before it is processed'(){
+		given:
+
+		when:
+			def response = appService.deleteResourceWithPassword(bobDunnBuddyConnectRequestDeleteURL, bobDunnPassword)
+
+		then:
+			response.status == 400
+			response.responseData?.code == "error.cannot.delete.unprocessed.message"
+	}
+
+	def 'Bob accepts Richard\'s buddy request'(){
+		given:
+
+		when:
+			def response = appService.postMessageActionWithPassword(bobDunnBuddyMessageAcceptURL, """{
+					"properties":{
+						"message":"Yes, great idea!"
+					}
+				}""", bobDunnPassword)
+
+		then:
+			response.status == 200
+			response.responseData.properties.status == "done"
+	}
+
+	def 'Richard checks his direct messages'(){
+		given:
+
+		when:
+			def response = appService.getDirectMessages(richardQuinURL, richardQuinPassword)
+			if (response.responseData._embedded && response.responseData._embedded.buddyConnectResponseMessages) {
+				richardQuinBuddyMessageProcessURL = response.responseData._embedded.buddyConnectResponseMessages[0]._links.process.href
+				richardQuinBuddyMessageDeleteURL = response.responseData._embedded.buddyConnectResponseMessages[0]._links.self.href
+			}
+
+		then:
+			response.status == 200
+			response.responseData._links.self.href == richardQuinURL + appService.DIRECT_MESSAGE_PATH_FRAGMENT
+			response.responseData._embedded.buddyConnectResponseMessages[0].user.firstName == "Bob ${timestamp}"
+			response.responseData._embedded.buddyConnectResponseMessages[0]._links.self.href.startsWith(response.responseData._links.self.href)
+			richardQuinBuddyMessageProcessURL.startsWith(response.responseData._embedded.buddyConnectResponseMessages[0]._links.self.href)
+			richardQuinBuddyMessageDeleteURL != null
+	}
+
+	def 'Richard tries to delete Bob\'s buddy acceptance before it is processed'(){
+		given:
+
+		when:
+			def response = appService.deleteResourceWithPassword(richardQuinBuddyMessageDeleteURL, richardQuinPassword)
+
+		then:
+			response.status == 400
+			response.responseData?.code == "error.cannot.delete.unprocessed.message"
```

Comment:
That still doesn't imply that you can delete it. The cleanup program should
take care of that.
On Nov 26, 2015 14:18, "aristotelos" <notifications@github.com> wrote:

> In
> appservice/src/intTest/groovy/nu/yona/server/DeleteDirectMessageTest.groovy
> <https://github.com/yonadev/yona-server/pull/29#discussion_r45977217>:
>
> > +			response.responseData._links.self.href == richardQuinURL + appService.DIRECT_MESSAGE_PATH_FRAGMENT
> > +			response.responseData._embedded.buddyConnectResponseMessages[0].user.firstName == "Bob ${timestamp}"
> > +			response.responseData._embedded.buddyConnectResponseMessages[0]._links.self.href.startsWith(response.responseData._links.self.href)
> > +			richardQuinBuddyMessageProcessURL.startsWith(response.responseData._embedded.buddyConnectResponseMessages[0]._links.self.href)
> > +			richardQuinBuddyMessageDeleteURL != null
> > +	}
> > +
> > +	def 'Richard tries to delete Bob\'s buddy acceptance before it is processed'(){
> > +		given:
> > +
> > +		when:
> > +			def response = appService.deleteResourceWithPassword(richardQuinBuddyMessageDeleteURL, richardQuinPassword)
> > +
> > +		then:
> > +			response.status == 400
> > +			response.responseData?.code == "error.cannot.delete.unprocessed.message"
>
> Maybe you would like to not reply to a connect request and let it expire
> instead of rejecting it. This is what I sometimes do with Facebook/LinkedIn
> to avoid disappointing people...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yonadev/yona-server/pull/29/files#r45977217>.
>